File: diagram-poc/frontend/src/app/core/services/comment.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { apiBaseUrl } from '../app-config';

export interface CommentItem {
  id: number;
  nodeId: string | null;
  authorName: string;
  text: string;
  createdAt: string;
  self: boolean;
}

const API = apiBaseUrl();

/** Comments on a diagram, optionally pinned to a block. */
@Injectable({ providedIn: 'root' })
export class CommentService {
  constructor(private http: HttpClient) {}

  list(diagramId: number): Observable<CommentItem[]> {
    return this.http.get<CommentItem[]>(`${API}/diagrams/${diagramId}/comments`);
  }

  add(diagramId: number, nodeId: string | null, text: string): Observable<CommentItem> {
    return this.http.post<CommentItem>(`${API}/diagrams/${diagramId}/comments`, { nodeId, text });
  }

  delete(commentId: number): Observable<void> {
    return this.http.delete<void>(`${API}/comments/${commentId}`);
  }
}

File: diagram-poc/backend/src/main/java/com/example/diagram/service/CommentService.java
package com.example.diagram.service;

import com.example.diagram.web.dto.CommentRequest;
import com.example.diagram.web.dto.CommentResponse;

import java.util.List;

/** Comments on diagrams (optionally pinned to a block). */
public interface CommentService {

    List<CommentResponse> list(Long diagramId, String userEmail);

    CommentResponse add(Long diagramId, CommentRequest request, String userEmail);

    void delete(Long commentId, String userEmail);
}
